src/ariel-os-sensors/src/sample.rs

```diff
@@ -43,8 +51,26 @@ impl Sample {
     }
 
     /// Returns the sample value.
+    ///
+    /// # Errors
+    ///
+    /// An error is returned if this value is temporarily unavailable (this part of the sensor is not ready).
+    pub fn value(&self) -> Result<i32, ValueError> {
+        match self.accuracy {
+            Accuracy::ValueTemporarilyUnavailable => Err(ValueError::TemporarilyUnavailable),
+            Accuracy::NoError
+            | Accuracy::Unknown
+            | Accuracy::SymmetricalError {
+                deviation: _,
+                bias: _,
+                scaling: _,
+            } => Ok(self.value),
```

Comment:
No need to list every field I think:
```suggestion
            | Accuracy::SymmetricalError { .. } => Ok(self.value),
```